### PR TITLE
[new] feat: add APL for NumberControl

### PR DESCRIPTION
**Pull request by aleareyes**
NumberControl APL cleanup, additional test cases, fixes for PR comments.
    
    - Removing validationFailedMessage dependency on values which are
    inaccessible at time of APL evaluation. Adding current value to
    the default APL validation failed message.
    - Default APL document cleanup.
    
    https://github.com/alexa/ask-sdk-controls/pull/33
## Description

Added APL support for NumberControl and default APL for entering a value via touch.

UX decisions taken:
*  If there is any APL version trouble (EditText), showing question/validation alone without keypad.
* Pressing the submit button on the keypad is the only way to send the event to the skill. Not supporting spoken "submit"
* For the default APL, putting the current value inside the failed validation message so that it's visible when making changes.

UX scenarios:
* If a value is entered via touch, no voice confirmation is required. If a value is given via voice, confirmation happens and user can confirm/disconfirm by voice or by providing a (same or corrected) value via touch.
* Users can resolve validation failures via any combination of voice and touch.

## Testing

Added demo with basic use case, unit tests covering default and custom cases.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs(Add new document content)
- [ ] Translate Docs(Translate document content)

## Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://https://github.com/alexa/ask-sdk-controls/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: src/commonControls/numberControl/NumberControl.ts [hunk 1]
```diff
@@ -590,6 +682,33 @@ export class NumberControl extends Control implements InteractionModelContributo
         }
     }
 
+    private isScreenEvent(input: ControlInput) {
```
Comment (Shreyas-vgr) [inline src/commonControls/numberControl/NumberControl.ts:685]: return type -> boolean

File: test/commonControlTests/NumberControl.spec.ts [hunk 2]
```diff
@@ -103,6 +158,46 @@ suite('NumberControl e2e tests', () => {
...
+                    TestInput.of(GeneralControlIntent.of({})),
+                    'A: What number?',
+                    'U: ',
+                    TestInput.simpleUserEvent(['NumberSelectorWithoutValidationExpectation', 15]),
+                    "A: Sorry but that's not a valid choice because the value must be even. What number?",
+                    'U: 16',
+                    TestInput.of(
+                        SingleValueControlIntent.of(AmazonBuiltInSlotType.NUMBER, { 'AMAZON.NUMBER': '16' }),
+                    ),
+                    'A: Ok. Value set to 16.',
```
Comment (Shreyas-vgr) [inline test/commonControlTests/NumberControl.spec.ts:194]: shouldn't it be `Was that 16?` to confirm since it was voice?

Comment (aleareyes) [reply test/commonControlTests/NumberControl.spec.ts:194]: In this case confirmation is only required if the value < 10 or > 20, but in general, yes we do confirm when values are given by voice (other tests verify this).

File: demo/NumberControl/NumberDemo/src/buildInteractionModel.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ * A copy of the License is located at
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import { NumberDemo1 } from '.';
```
[the remark this answers is not in the document]
Comment (aleareyes) [reply demo/NumberControl/NumberDemo/src/buildInteractionModel.ts:14]: Hm, this was just consistency with ListControl Demo: https://github.com/alexa/ask-sdk-controls/blob/develop/demo/ListControl/YesNoMaybe/src/index.ts#L18

But I do agree, will change it to BasicNumberDemo.

**Review by mikelid — APPROVED** [review]

**Review by Shreyas-vgr — APPROVED** [review]

**Pull request merged**